src/polygon/CMakeLists.txt

```diff
@@ -15,7 +15,7 @@
 project (polygon)
 
 file(GLOB_RECURSE SOURCE_FILES CONFIGURE_DEPENDS *.cpp)
-add_library(${PROJECT_NAME} ${SOURCE_FILES})
+add_library(${PROJECT_NAME} OBJECT ${SOURCE_FILES})
```

Comment:
Since it is a sublibrary not used on it's own, this lets it compile separately still, but get linked into the main `a.`/`.so` being built. I was having some weird issues with relocating the many archives/objects when bundling for the ocaml library, and doing this to get rid of the extra library artifacts fixed it. The [entry](https://cmake.org/cmake/help/latest/command/add_library.html#object-libraries) in the cmake manual about it is most of what I know about OBJECT.